src/ray/gcs/pubsub/gcs_pub_sub.h

```diff
@@ -264,9 +265,14 @@ class GcsSubscriber {
   GcsSubscriber(const std::shared_ptr<RedisClient> &redis_client,
                 const rpc::Address &gcs_address,
                 std::unique_ptr<pubsub::Subscriber> subscriber)
-      : pubsub_(std::make_unique<GcsPubSub>(redis_client)),
-        gcs_address_(gcs_address),
-        subscriber_(std::move(subscriber)) {}
+      : gcs_address_(gcs_address), subscriber_(std::move(subscriber)) {
+    if (redis_client) {
```

Comment:
Why not keep the initialization logic as before, and only check when
```
if (pubsub_ != nullptr) {
  RAY_CHECK(...) << ...;
}
```

Reply:
Once gcs address is setup, I don't want the old path to be used anymore.

btw, what do you mean by `if (pubsub_ != nullptr) {` ? It's in the constructor and it's not initialized right?

Reply:
Ok, I guess we needed to change the initialization logic to keep pubsub_ const. Leaving it non-const is fine.